### ISSUE TITLE
Ray start script setting `--include-dashboard=false` does not work

### ISSUE THREAD

**Issue by WeichenXu123** [ray-project/ray#31465]
### What happened + What you expected to happen

Runs command:
```
ray start --head --block --include-dashboard=false
```

It still starts ray dashboard service,
and
```
  To monitor and debug Ray, view the dashboard at 
    127.0.0.1:8265
```
is printed,
and run `curl 127.0.0.1:8265` gets feedback from the dashboard service.

### Versions / Dependencies

master version.

### Reproduction script

```
ray start --head --block --include-dashboard=false
```

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

**Comment by rkooo567** [comment]
Hmm it is pretty surprising.. we will fix this asap

**Comment by jovany-wang** [comment]
Hum, Yes. we also found this issue these days. :(
Looking forward the fix. @rkooo567

**Comment by WeichenXu123** [comment]
@rkooo567 Any progress on the fix ? Thank you!

**Comment by rkooo567** [comment]
not yet! I will fix the issue by ray 2.3 (planned to be release in the middle of feb)!

**Comment by rkooo567** [comment]
Planning to fix tomorrow. I have a dedicated time to work on bug fix tmrw

**Comment by cointreau** [comment]
I had same problem too. My ray version is 2.2.
I run with `ray.init(num_cpus=2, log_to_driver=False, include_dashboard=False)` and get messages `Started a local Ray instance. view the dashboard at 127.0.0.1:8265...`

One weird thing was, this problem did not happen before.
When I set ray with command `pip install ray`, and it worked well.
But, when I tried `ray memory` command, I got a messages that I need to get aiohttp, opencensus, prometheus_client, aiohttp_cors and etc... so I installed them as well.

After installing them, `ray memory` works but include_dashboard=False option does not work.

I removed all of packages newly installed, so now the option works, but I can't use `ray memory`.....

**Comment by rkooo567** [comment]
I made a PR ^.

Note that this only started a HTTP server, but it disabled all the features (e.g., data aggregation or serving). It is the same as the dashboard is not enabled.

My PR also disables the HTTP server so you won't even be able to access the frontend after it is merged.

**Comment by WeichenXu123** [comment]
Thank you @rkooo567 !

**Comment by jovany-wang** [comment]
@rkooo567 Let me have a check: this commit will be released in 2.3 or 2.4?

**Comment by rkooo567** [comment]
This must be in 2.3
